### PR TITLE
[plugin.video.tv3.cat] 1.4.3

### PR DESCRIPTION
**Pull request by jqandreu**
### Description
Code redesigned to classes and migrated to Python 2/3

### Checklist:

- [x ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x ] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.tv3.cat/resources/lib/ui/UI.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+            mode = folder.mode
+            name = folder.name
+            nameQuoted = urllib.parse.quote(name)
+            url = folder.url
+            iconImage = folder.iconImage
+            thumbImage = folder.thumbnailImage
+
+            urlPlugin = buildUrl({'mode': mode, 'name': nameQuoted, 'url': url}, self.base_url)
+            liz = xbmcgui.ListItem(name, iconImage=iconImage, thumbnailImage=thumbImage)
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/ui/UI.py]:
```python
liz = xbmcgui.ListItem(name)
liz.setArt({'thumb': thumbImage, 'icon': iconImage})
```

File: plugin.video.tv3.cat/resources/lib/ui/UI.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                    urlVideo = video.url
+                    xbmc.log("UI - listVideos - urlVideo: " + urlVideo)
+                    iconImage = video.iconImage
+                    thumbImage = video.thumbnailImage
+                    durada = video.durada
+                    titol = video.title
+
+                    urlPlugin = buildUrl({'mode':'playVideo','name':"",'url':urlVideo}, self.base_url)
+
+                    liz = xbmcgui.ListItem(titol, iconImage="DefaultVideo.png", thumbnailImage=thumbImage)
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/ui/UI.py]: same

File: plugin.video.tv3.cat/resources/lib/ui/UI.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+            if last:
+                mode = last.mode
+                name = last.name
+                url = last.url
+                #xbmc.log("UI - listVideos - urlNext: " + url)
+                iconImage = last.iconImage
+                thumbImage = last.thumbnailImage
+
+                urlPlugin = buildUrl({'mode': mode, 'name': '', 'url': url}, self.base_url)
+                liz = xbmcgui.ListItem(name, iconImage=iconImage, thumbnailImage=thumbImage)
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/ui/UI.py]: same

File: plugin.video.tv3.cat/resources/lib/utils/Utils.py [hunk 4]
```diff
@@ -1,97 +1,91 @@
...
- @author: jqandreu
- @contact: jqsandreu@gmail.com
-'''
-
-import urllib
-import urllib2
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str
+import urllib.request, urllib.parse, urllib.error
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/utils/Utils.py]: not py2compatible

File: plugin.video.tv3.cat/resources/lib/utils/Utils.py [hunk 5]
```diff
@@ -1,97 +1,91 @@
...
+
         return link
-    
-    except urllib2.URLError as e:
+
+    except urllib.error.URLError as e:
         xbmc.log("getHtml error - " + str(e))
         xbmc.log("getHtml url - " + url)
-        
+        xbmcgui.Dialog().notification("Error a la connexio", "No s'ha pogut connectar", xbmcgui.NOTIFICATION_INFO)
```
Comment (enen92) [inline plugin.video.tv3.cat/resources/lib/utils/Utils.py]: These strings need to be localized

File: plugin.video.tv3.cat/addon.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
+from __future__ import unicode_literals
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str
+import sys
+import urllib.parse
```
[the remark this answers is not in the document]
Comment (jqandreu) [reply plugin.video.tv3.cat/addon.py]: This imports are from future module to have compatibility with Python 2 and Python 3. I don't understand why this is not compatible with python 2. The plugin works with kodi v17 and python 2.7

File: plugin.video.tv3.cat/resources/lib/utils/Utils.py (hunk 4 above)
Comment (jqandreu) [reply plugin.video.tv3.cat/resources/lib/utils/Utils.py]: Also there are imports form future script

File: plugin.video.tv3.cat/addon.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (enen92) [reply plugin.video.tv3.cat/addon.py]: Yeap, ignore. Realised that and forgot to comment again.

**Comment by jqandreu** [comment]
I have fixed the requested changes. Thanks

File: plugin.video.tv3.cat/resources/lib/ui/UI.py (hunk 1 above)
Comment (jqandreu) [reply plugin.video.tv3.cat/resources/lib/ui/UI.py]: Fixed

File: plugin.video.tv3.cat/addon.xml [hunk 7]
```diff
@@ -1,10 +1,11 @@
...
     <requires>
         <import addon="xbmc.python" version="2.14.0"/>
 		<import addon="script.module.beautifulsoup" version="3.2.1"/>
+        <import addon="script.module.future" version="0.17.1"/>
     </requires>
-    <extension point="xbmc.python.pluginsource" library="default.py">
+    <extension point="xbmc.python.pluginsource" library="addon.py">
         <provides>video</provides>
     </extension>
     <extension point="xbmc.addon.metadata">
```
[the remark this answers is not in the document]
Comment (jqandreu) [reply plugin.video.tv3.cat/addon.xml]: Fixed

File: plugin.video.tv3.cat/.gitignore [hunk 8]
```diff
@@ -217,3 +217,9 @@ pip-log.txt
 
 #Git
```
[the remark this answers is not in the document]
Comment (jqandreu) [reply plugin.video.tv3.cat/.gitignore]: Fixed

File: plugin.video.tv3.cat/resources/lib/ui/UI.py (hunk 1 above)
Comment (jqandreu) [reply plugin.video.tv3.cat/resources/lib/ui/UI.py]: Fixed

**Comment by enen92** [comment]
Thanks for adjusting

**Pull request merged**